sr_utilities_common/src/sr_utilities_common/aws_manager.py

```diff
@@ -75,62 +73,187 @@ def get_bucket_structure_with_prefix(self, bucket_name, prefix):
+        if bucket_subfolder:
+            print_msg += f"\n    {bucket_subfolder}/{file}"
+        else:
+            print_msg += f"\n    {file}"
+    print_msg += f"\n\nThese files will be downloaded too {files_base_path}/{files_folder_path}:"
+    print_msg += print_msg_2
+    rospy.loginfo(print_msg)
+    value = input("Would you like to download these files? (Y/N) ")
+    value = value.lower().strip()
+    if value == "y":
+        return True
+    elif value == "n":
+        return False
+    else:
+        rospy.logerr("Select a valid option")
+        rospy.signal_shutdown("")
+        exit(1)
+
+
+def validated_files_to_be_uploaded(bucket_name, files_base_path, files_folder_path,
+                                   file_names, bucket_subfolder):
+    print_msg = f"\nFrom {files_base_path}/{files_folder_path} uploading the files:"
+    print_msg_2 = ""
+    for file in file_names:
+        print_msg += f"\n    {files_folder_path}/{file}"
+        if bucket_subfolder:
+            print_msg_2 += f"\n    {bucket_subfolder}/{file}"
+        else:
+            print_msg_2 += f"\n    {file}"
+    print_msg += f"\n\nThese files will be uploaded to the bucket {bucket_name}:"
+    print_msg += print_msg_2
+    rospy.loginfo(print_msg)
+    value = input("Would you like to upload these files? (Y/N) ")
+    value = value.lower().strip()
+    if value == "y":
+        return True
+    elif value == "n":
+        return False
+    else:
+        rospy.logerr("Select a valid option")
+        rospy.signal_shutdown("")
+        exit(1)
+
+
+def return_function_mode(function_mode):
+    function_mode_edited = function_mode.lower().strip()
+    if function_mode_edited != "download" and function_mode_edited != "upload":
+        err_message = "Please specify either upload or download to use this script"
+        err_message += "\nE.g: function_mode:='upload'"
+        err_message += "\n     function_mode:='download'"
+        err_message += f"You entered: {function_mode}"
+        rospy.logerr(err_message)
+        rospy.signal_shutdown("")
+        exit(1)
+    return function_mode_edited
+
+
 if __name__ == "__main__":
     rospy.init_node("aws_manager_node")
 
-    download_param = rospy.get_param("~download")
-    upload_param = rospy.get_param("~upload")
+    function_mode = rospy.get_param("~function_mode")
+    skip_check = rospy.get_param("~skip_check")
     bucket_name = rospy.get_param("~bucket_name")
+    bucket_subfolder = rospy.get_param("~bucket_subfolder")
     files_base_path = rospy.get_param("~files_base_path")
     files_folder_path = rospy.get_param("~files_folder_path")
     file_names = rospy.get_param("~file_names")
 
+    function_mode = return_function_mode(function_mode)
+    if bucket_subfolder == "":
+        bucket_subfolder = None
     aws_manager = AWS_Manager()
 
-    if upload_param is True:
+    if function_mode == "upload":
+        if file_names == "":
```

Comment:
Here you're checking for file_names but not within the _prepare_structure_download. Maybe it would be good to check it (with the gather_all_files_local method) already inside the above method?